File: dms-frontend/src/app/core/services/profile.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiResponse } from '../../shared/models/api-response.model';
import { ApiService } from './api.service';

export interface ProfileData {
  id: number;
  username: string;
  email: string;
  phoneNumber: string;
  fullName: string;
  dateOfBirth: string;
  gender: string;
  address?: string;
  block?: string;
  town?: string;
  state?: string;
  village?: string;
  landmark?: string;
  district?: string;
  country?: string;
  pinCode?: string;
  registrationMethod: string;
  profilePhotoUrl: string;
  paymentStatus: string;      // 'PENDING' | 'COMPLETED'
  profileStatus: string;      // 'Active' | 'Inactive'
  isActive: boolean;          // true only after admin approval
  registrationStatus: string;
  createdAt: string;
  updatedAt: string;
  zone?: string;
  referralCode?: string;
  referredByUsername?: string;
}

@Injectable({
  providedIn: 'root'
})
export class ProfileService {

  constructor(private readonly apiService: ApiService) {}

  getMyProfile(): Observable<ApiResponse<ProfileData>> {
    return this.apiService.getMyProfile();
  }

  updateProfile(formData: FormData): Observable<ApiResponse<ProfileData>> {
    return this.apiService.updateProfile(formData);
  }

  generateReferralCode(): Observable<ApiResponse<ProfileData>> {
    return this.apiService.generateReferralCode();
  }
}

File: dms/src/main/java/com/example/documentmanagement/service/ProfileService.java
package com.example.documentmanagement.service;

import com.example.documentmanagement.dto.response.ProfileResponse;
import org.springframework.web.multipart.MultipartFile;

public interface ProfileService {
    ProfileResponse getMyProfile(String username);
    ProfileResponse updateProfile(String username, String fullName, String gender, String dateOfBirth, 
                                  String address, String block, String town, String state, String village, String landmark,
                                  String district, String country, String pinCode, String zone, MultipartFile photo);
    ProfileResponse completePayment(String username);
    ProfileResponse generateReferralCode(String username);
}
